OpenRA.Mods.Common/Traits/BotModules/UnitBuilderBotModule.cs

```diff
@@ -0,0 +1,203 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Controls AI unit production.")]
+	public class UnitBuilderBotModuleInfo : ConditionalTraitInfo
+	{
+		public readonly HashSet<string> McvTypes = new HashSet<string>();
+		public readonly HashSet<string> ConstructionYardTypes = new HashSet<string>();
+		public readonly HashSet<string> McvFactoryTypes = new HashSet<string>();
+		public readonly HashSet<string> RefineryTypes = new HashSet<string>();
+		public readonly HashSet<string> PowerTypes = new HashSet<string>();
```

Comment:
I'm hoping that most of these can go away from here after implementing the interface suggestions below: